src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/Microsoft.VisualStudio.LanguageServerClient.Razor.csproj

```diff
@@ -19,6 +19,9 @@
   <ItemGroup>
     <PackageReference Include="System.Composition" Version="$(SystemCompositionPackageVersion)" />
     <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="$(Tooling_MicrosoftCodeAnalysisAnalyzersPackageVersion)" NoWarn="NU1608" />
+    <PackageReference Include="Microsoft.Extensions.Caching.Abstractions" Version="$(MicrosoftExtensionsCachingAbstractionsPackageVersion)" />
```

Comment:
Ya I'm confused, the dependency chain for Extensions.Caching.Memory:
![image](https://user-images.githubusercontent.com/2008729/87963860-402b1500-ca6e-11ea-8e95-a6e3917df98c.png)

It contains the abstractiosn and the primitives. I don't see why you'd need to reference these packages (i'd imagine they'd be brought in transitively). I'd still imagine you'd need to provide a codebase for all 3 but not reference all 3 as packages (unless there was a dependency conflict).

Reply:
I had tried that (commit: 57608ab44), however I still got the same errors:
```
System.AggregateException: Project system data flow 'CapabilitiesScope Aggregating Publishing: 51635752' closed because of an exception.
An exception was thrown while initializing part "Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector". ---> Microsoft.VisualStudio.Composition.CompositionFailedException: An exception was thrown while initializing part "Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector". ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.Caching.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
```
This is with the `Assembly Bases` [still having](https://github.com/dotnet/aspnetcore-tooling/blob/57608ab449c93c1ccea95a0a5f34560ee6ca810d/src/Razor/src/Microsoft.VisualStudio.RazorExtension/AssemblyCodeBases.cs#L13-L15) the primitives/abstractions dependencies.

Reply:
The hell..... In that case is the Extensions.Caching.Abstractions dll in the RazorExtension output directory? Aka are the codebases actually pointing to an existent assembly

Reply:
With https://github.com/dotnet/aspnetcore-tooling/commit/57608ab449c93c1ccea95a0a5f34560ee6ca810d Abstractions/Primitives are in:

`artifacts\bin\Microsoft.VisualStudio.RazorExtension\Debug\net472`

with the expected 5-preview-8 version:
![image](https://user-images.githubusercontent.com/14852843/87967671-06f5a380-ca74-11ea-95a1-2bf14d746356.png)


but 

```
System.AggregateException: One or more errors occurred. ---> Microsoft.VisualStudio.Composition.CompositionFailedException: An exception was thrown while initializing part "Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector". ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.Extensions.Caching.Abstractions, Version=5.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60' or one of its dependencies. The system cannot find the file specified.
   at Microsoft.VisualStudio.LanguageServerClient.Razor.DefaultLSPEditorFeatureDetector..ctor(ProjectHierarchyInspector projectHierarchyInspector)
```

persists.

Reply:
Resolved with https://github.com/dotnet/aspnetcore-tooling/pull/2235/commits/7039a27e0d67fcc6b7130e346e48296ffd042f97 (needed to be included as a VSIX Source). Thanks @NTaylorMullen!